File: java/java-arithmetic/src/main/java/com/leetcode/first/J104.java
package com.leetcode.first;

public class J104 {

    public static void main(String[] args) {

    }

    public int maxDepth(TreeNode root) {
        if ( root ==null){
            return 0;
        }
        int leftDep=maxDepth(root.left);
        int rightDep=maxDepth(root.right);
        return Math.max(leftDep,rightDep)+1;
    }
}

File: test-python/leetcode/J104.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        if not root:
            return 0

        leftMaxDep=self.maxDepth(root=root.left)
        rightMaDep=self.maxDepth(root=root.right)

        return max(leftMaxDep,rightMaDep)+1

